### PR TITLE
feat: applicationset volumes and volumemounts spec

### PR DESCRIPTION
**Pull request by appiepollo14**
What type of PR is this?

/kind enhancement

What does this PR do / why we need it:

To be able to add extra volumes and volumeMounts to the appSet deployment, see related issue.

Have you updated the necessary documentation?

 Documentation update is required by this PR.
 Documentation has been updated.
Which issue(s) this PR fixes:

Fixes https://github.com/redhat-developer/gitops-operator/issues/620

How to test changes / Special notes to the reviewer:

### CODE REVIEW

**Comment by appiepollo14** [comment]
@svghadi can you please help me with the following questions:
a. Is there a need to write e2e-tests for this quite trivial change. I've written unit tests for it.
b. If e2e-tests are required, I've added a starter, can you explain why it if failing. Check the CI-logs. How is it possible that the modified applicationSet api, is not available in the e2e tests. Is this by design? Can't a api change be accompanied with a new e2e-test in the same PR? Please give me some pointers. 

see newly created issue https://github.com/argoproj-labs/argocd-operator/issues/1566 for running this locally. Would be very handy to be able to test the kuttl tests locally. Would decrease a cycle with 45 mins in this case.

**Comment by appiepollo14** [comment]
@svghadi ready for your approval!

**Comment by svghadi** [comment]
> a. Is there a need to write e2e-tests for this quite trivial change. I've written unit tests for it.

Well, it depends on the feature. Personally, I try to add e2e tests for every feature. However, sometimes it can be tricky, so exceptions can be made.

> Edit: I've come to the conclusion that I've was using the v1alpha1 api in the e2e-test and changed the v1beta1 api. Which raises my question. Is it OK to only change v1beta1 spec?

Yes. v1alpha1 is deprecated so changes to v1beta1 only are fine.

**Review by svghadi — CHANGES_REQUESTED** [review]
Left a suggestion, rest all looks good. 

Also, it would be good to have a unit or e2e test to check default vols & mounts when additional vols & mounts are not set/removed from ArgoCD CR.

File: tests/k8s/1-007_validate_volume_mounts/02-assert.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+---
```
[the remark this answers is not in the document]
Comment (appiepollo14) [reply tests/k8s/1-007_validate_volume_mounts/02-assert.yaml:8]: Done!

**Comment by appiepollo14** [comment]
> Left a suggestion, rest all looks good.
> 
> Also, it would be good to have a unit or e2e test to check default vols & mounts when additional vols & mounts are not set/removed from ArgoCD CR.

There already is, each test calling: https://github.com/appiepollo14/argocd-operator/blob/feat/appsetcontroller-extra-volume-mounts/controllers/argocd/applicationset_test.go#L106 checks the default vols * mounts.

**Comment by svghadi** [comment]
Okay. Looks good then. Can you fix the merge conflicts?

**Comment by appiepollo14** [comment]
> Okay. Looks good then. Can you fix the merge conflicts?

Fixed!

**Review by svghadi — APPROVED** [review]
LGTM.
Thanks @appiepollo14 for the contribution.

**Pull request merged**